### DISCUSSION TITLE
Property testing

### DISCUSSION THREAD

**Discussion by tom-sherman** [koka-lang/koka#296] [General]
Has there been any experiments with property testing in koka?

One thing I've found annoying about property testing is that it's never very ergonomic to use, especially with custom arbitraries. Threading the random state through is awkward.

This complaint is a variation on "monads are awkward" so I'm sure effects can help here.

I've started playing around with this idea over at https://github.com/tom-sherman/fizzy.kk

**Comment by Mazawaje** [comment]
Good job
